src/backend/plugins/organizations.py

```diff
@@ -33,20 +34,17 @@ def _extract_name_from_organization_data(organization_data):
             logger.warning("Empty list 'liste_enseignes' in %s", organization_data)
         return None
 
-    def _get_organization_name_from_siret(self, siret):
-        """Return the organization name from the SIRET."""
-        try:
-            response = requests.get(self._api_url.format(siret=siret), timeout=10)
-            response.raise_for_status()
-            data = response.json()
-        except requests.RequestException as exc:
-            logger.exception("%s: Unable to fetch organization name from SIRET", exc)
-            return None
-
+    def get_organization_name_from_results(self, data, siret):
+        """Return the organization name from the results of a SIRET search."""
         for result in data["results"]:
+            nature = "nature_juridique"
+            commune = nature in result and result[nature] == "7210"
             for organization in result["matching_etablissements"]:
                 if organization.get("siret") == siret:
-                    return self._extract_name_from_organization_data(organization)
+                    if commune:
+                        return organization["libelle_commune"].title()
+
+                return self._extract_name_from_organization_data(organization)
```

Comment:
Warning, the indentation has changed here, this might get the wrong "siret".

Reply:
I'm not sure I see how to construct a case where we'd get the wrong result here (i.e. a missing unit test). The indent is by intention: I did mean to return directly with the `libelle_commune` in the commune case (because I'm pretty sure the key is always present) and fall through to the invocation of the private method otherwise. (So we end up with a bit of asymmetry in the code with some error handling in one case and not in the other.)